flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/BufferManager.java

```diff
@@ -208,7 +215,13 @@ public void recycle(MemorySegment segment) {
         }
 
         try {
-            inputChannel.notifyBufferAvailable(numAddedBuffers);
+            if (releasedFloatingBuffer == null) {
+                if (!inputChannel.isReleased()) {
+                    inputChannel.notifyBufferAvailable(1);
+                }
+            } else {
+                releasedFloatingBuffer.recycleBuffer();
+            }
         } catch (Throwable t) {
             ExceptionUtils.rethrow(t);
         }
```

Comment:
I'm a little bit worried about exception handling. I think the code as it is works fine, but might be giving a wrong impression, that it's trying to support exception handling when recycling the `releaseFloatingBuffer`, while in reality, if any exception happens on the code path on which we are trying to recycle this floating buffer, it will be leaked/lost.

Maybe this would be reflect this better?
```
if (releasedFloatingBuffer != null) {
    releasedFloatingBuffer.recycleBuffer();
}
else {
    try {
        if (!inputChannel.isReleased()) {
            inputChannel.notifyBufferAvailable(1);
        }
    } catch (Throwable t) {
        ExceptionUtils.rethrow(t);
    }
}
```

Reply:
Yes it is!